benchmark/bq/harness.js

```diff
@@ -1,26 +1,57 @@
-var Queue = require('../../index');
-var queue = new Queue('test');
+const helpers = require('../../lib/helpers');
+const Queue = require('../..');
+const queue = new Queue('test', process.env.BQ_MINIMAL ? {
+  storeJobs: false,
+  sendEvents: false,
+  getEvents: false,
+  removeOnSuccess: true
+} : {
+  removeOnSuccess: true
+});
 
-module.exports = function (options, cb) {
-  var finished = 0;
-  var finishTime, startTime;
-
-  var reportResult = function (result) {
-    finished += 1;
-    if (finished === options.numRuns) {
-      finishTime = (new Date()).getTime();
-      cb(null, finishTime - startTime);
+// A promise-based barrier.
+function reef(n = 1) {
+  const done = helpers.deferred(), end = done.defer();
+  return {
+    done,
+    next() {
+      --n;
+      if (n < 0) return false;
+      if (n === 0) end();
+      return true;
     }
   };
+}
 
-  queue.process(options.concurrency, function (job, done) {
-    reportResult();
-    return done();
-  });
+module.exports = (options) => {
```

Comment:
Let's similarly wrap this in a `queue.ready().then(...)` to factor out startup time.

Reply:
Fixed

Reply:
Fixed